core/src/main/scala/chisel3/experimental/package.scala

```diff
@@ -130,6 +132,32 @@ package object experimental {
     }
   }
 
+  /** This class provides the `Lit` method needed to define a `Vec` literal
+    */
+  object VecLiterals {
+    implicit class AddVecLiteralConstructor[T <: Vec[_]](x: T) {
+      /** Given a generator of a list tuples of the form [Int, Data]
+        * constructs a Vec literal, parallel concept to `BundleLiteral`
+        *
+        * @param elems generates a lit of `(Int, Data)` where the I
+        * @return
+        */
+      def Lit(elems: (Int, Data)*): T = {
+        x._makeLit(elems: _*)
+      }
+    }
+
+    implicit class AddObjectLiteralConstructor[T <: Vec.type](x: T) {
+      def Lit[T <: Data](elems: T*)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): Vec[T] = {
+        require(elems.nonEmpty, s"Lit.Vec(...) must have at least one element")
```

Comment:
This seems like a reasonable require, but as people start writing generators with this they may well want a zero length literal. Might need semantics like `Vec[Nothing]` which may not be well supported. But it'll be a strict API expansion.

Reply:
You are probably right that generators should cover a zero length. Do you have a suggestion as to what should happen in this case? Because this is all experimental, it seems like changing this small aspect of the API can be reasonably changed in the future.

Reply:
Note that we currently don't allow empty `VecInit()`, I see this as related. We *should* support both but I think having this consistent is reasonable and when we figure out how to add support we should add to both.